lib/moonshot/command_line.rb

```diff
@@ -0,0 +1,121 @@
+require 'thor'
+
+module Moonshot
+  # This class implements the command-line `moonshot` tool.
+  class CommandLine
+    def self.run! # rubocop:disable AbcSize, CyclomaticComplexity, MethodLength, PerceivedComplexity
+      # Find the Moonfile in this project.
+      orig_dir = Dir.pwd
+
+      loop do
+        break if File.exist?('Moonfile.rb')
+        raise 'Could not find Moonfile.rb!' if Dir.pwd == '/'
+
+        Dir.chdir('..')
+      end
+
+      moonfile_dir = Dir.pwd
+      Dir.chdir(orig_dir)
+
+      # Load any plugins and CLI extensions relative to the Moonfile
+      if File.directory?(File.join(moonfile_dir, 'moonshot'))
+        load_cli_extensions(moonfile_dir)
+        load_plugins(moonfile_dir)
+      end
+
+      Object.include(Moonshot::ArtifactRepository)
+      Object.include(Moonshot::BuildMechanism)
+      Object.include(Moonshot::DeploymentMechanism)
```

Comment:
I never noticed this before. Is this to shorten the constant lookup in the Moonfile?

If so, do you only need that to happen inside the config block? Just wondering about a solution that wouldn't touch Object, but it's not critical.